src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -175,20 +184,51 @@ def check_cluster_connection(self, url, payload, username, ssl_verify, ssl_certi
 
         cluster_token = content['token']
 
+        if cluster_token:
+            self.check_connection_errors(url, cluster_token, ssl_verify, ssl_certificate, action)
+        return cluster_token
+
+    def check_connection_errors(self, url, cluster_token, ssl_verify, ssl_certificate, action):
         managed_by_clusters_content = self._proxy('GET', url, 'api/settings/MANAGED_BY_CLUSTERS',
                                                   token=cluster_token, verify=ssl_verify,
                                                   cert=ssl_certificate)
 
         managed_by_clusters_config = managed_by_clusters_content['value']
 
-        if len(managed_by_clusters_config) > 1:
+        if len(managed_by_clusters_config) > 1 and action == 'connect':
             raise DashboardException(msg='Cluster is already managed by another cluster',
                                      code='cluster_managed_by_another_cluster',
                                      component='multi-cluster')
-        return cluster_token
+
+        self.check_security_config(url, cluster_token, ssl_verify, ssl_certificate)
+
+    def check_security_config(self, url, cluster_token, ssl_verify, ssl_certificate):
+        remote_security_config_content = self._proxy('GET', url,
```

Comment:
`remote_security_config_content` variable name is too long IMHO, what about `remote_security_cfg` ?

Reply:
Updated